### PR TITLE
[Tabs] Add titleColorForState: API.

### PR DESCRIPTION
**Pull request by romoore**
Adds a new API to affect the items' title color for `.selected` and `.normal` states.

Part of #7657

### CODE REVIEW

File: components/Tabs/src/TabBarView/MDCTabBarView.m [hunk 1]
```diff
@@ -21,7 +21,10 @@
 @interface MDCTabBarView ()
 
 /** The views representing the items of this tab bar. */
-@property(nonatomic, strong) NSArray<UIView *> *itemViews;
+@property(nonnull, nonatomic, strong) NSArray<UIView *> *itemViews;
+
+/** The title colors for bar items. */
+@property(nonnull, nonatomic, strong) NSMutableDictionary<NSNumber *, UIColor *> *titleColors;
```
Comment (bryanoltman) [inline components/Tabs/src/TabBarView/MDCTabBarView.m]: It was not immediately clear to me that this dictionary is mapping `UIControlState`s to `UIColor`s. Please update either the comment or the property name to reflect this. I'm not sure what our naming conventions are for dictionaries, but something like `statesToTitleColors` would be a lot clearer.

File: components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m [hunk 2]
```diff
@@ -163,4 +163,59 @@ - (void)testSettingBackgroundColorUpdatesBarTintColor {
...
+}
+
+- (void)testTitleColorForStateWithNoValuesReturnsNil {
+  // When
+  [self.tabBarView setTitleColor:nil forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:nil forState:UIControlStateSelected];
+
+  // Then
+  XCTAssertNil([self.tabBarView titleColorForState:UIControlStateNormal]);
+  XCTAssertNil([self.tabBarView titleColorForState:UIControlStateSelected]);
```
Comment (bryanoltman) [inline components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m:218]: What is the expected behavior if a title color is set for a non-normal control state, but the title color for `UIControlStateNormal` is `nil`?

**Review by bryanoltman — CHANGES_REQUESTED** [review]

File: components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m (hunk 2 above)
Comment (romoore) [reply components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m:218]: "A default color is ~returned~ used." The _returned_ value will be `nil`.

File: components/Tabs/src/TabBarView/MDCTabBarView.m (hunk 1 above)
Comment (romoore) [reply components/Tabs/src/TabBarView/MDCTabBarView.m]: Done.

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**